pkg/codegen/dotnet/gen_program_expressions.go

```diff
@@ -618,9 +619,15 @@ func (g *generator) GenFunctionCallExpression(w io.Writer, expr *model.FunctionC
 	case "fromBase64":
 		g.Fgenf(w, "System.Text.Encoding.UTF8.GetString(Convert.FromBase64String(%v))", expr.Args[0])
 	case "toJSON":
-		g.Fgen(w, "JsonSerializer.Serialize(")
-		g.genDictionaryOrTuple(w, expr.Args[0])
-		g.Fgen(w, ")")
+		if model.ContainsOutputs(expr.Args[0].Type()) {
+			g.Fgen(w, "Output.JsonSerialize(Output.Create(")
```

Comment:
We need `Output.JsonSerialize(Output.Create(...))` instead of `Output.JsonSerialize(...)` because the function only accepts `Output<T>`. We could simplify that in a subsequent PR after we generalize `Output.JsonSerialize` to accept any `T`